src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -5565,3 +5565,65 @@ void C2_MacroAssembler::udivmodL(Register rax, Register divisor, Register rdx, R
 }
 #endif
 
+void C2_MacroAssembler::rearrange_bytes(XMMRegister dst, XMMRegister shuffle, XMMRegister src, XMMRegister xtmp1,
```

Comment:
Can we use the same approach as that used for 256-bit vector. Something similar to:

    vpshufb(xtmp1, src, shuffle); // All elements are at the correct place modulo 16
    vpxor(dst, dst, dst);
    vpslld(xtmp2, shuffle, 3); // Push the digit signifying the parity of 128-bit lane to the sign digit
    vpcmpb(ktmp, xtmp2, dst, le);
    vshufi32x4(xtmp2, xtmp1, xtmp1, 0b10110001); // Shuffle the 128-bit lanes to get 1 - 0 - 3 - 2
    vpblendmb(xtmp1, ktmp, xtmp1, xtmp2); // All elements are at the correct place modulo 32
    vpslld(xtmp2, shuffle, 2); // Push the digit signifying the parity of 256-bit lane to the sign digit
    vpcmpb(ktmp, xtmp2, dst, le);
    vshufi32x4(xtmp2, xtmp1, xtmp1, 0b01001110); // Shuffle the 128-bit lanes to get 2 - 3 - 0 - 1
    vpblendmb(dst, ktmp, xtmp1, xtmp2); // All elements are at the correct place modulo 64

Reply:
Actually, it is my bad, this should not work. Sorry for the noise.